hipify caffe2 script in fbcode

xw285cornell opened the pull request:
Summary:
Make changes to make hipify_python script to work with fbcode.

1. Add TARGETS file
2. Make hipify_python a module as well as a standalone script.

Differential Revision: D10851216

bddppq commented on tools/amd_build/build_pytorch_amd.py:
```diff
@@ -1,9 +1,7 @@
-import sys
-from shutil import copytree, ignore_patterns
 from functools import reduce
+from pyHIPIFY import hipify_python
```
This probably doesn't work with python3 in OSS, maybe better to add sys.path

bddppq commented on tools/amd_build/pyHIPIFY/hipify_python.py:
```diff
@@ -45,6 +45,33 @@
 CAFFE2_TEMPLATE_MAP = {}
 
 
+class Error(Exception):
```
maybe simply 
```
class Error(Exception): pass
class InputErorr(Error): pass
```
?

bddppq commented on tools/amd_build/pyHIPIFY/hipify_python.py:
```diff
@@ -45,6 +45,33 @@
+class Error(Exception):
+    """Base class for exceptions in this module."""
+
+    def __init__(self, expression, message, desc):
```
The only place uses InputError is only passing one argument:
https://github.com/pytorch/pytorch/blob/d05b4e31d1a41bf5c908adb2d3a0d4d6ba0118fc/tools/amd_build/pyHIPIFY/hipify_python.py#L422

bddppq commented on tools/amd_build/pyHIPIFY/hipify_python.py:
```diff
@@ -695,7 +727,7 @@ def disable_function(input_string, function, replace_style):
 
 
-def get_hip_file_path(filepath, hipify_caffe2):
+def get_hip_file_path(filepath, hipify_caffe2, hip_suffix):
```
I suggest let's use ".hcc" everywhere

xw285cornell replied:
Yes I'll change to hcc everywhere :)

xw285cornell committed:
hipify caffe2 script in fbcode (#13265)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/13265

Make changes to make hipify_python script to work with fbcode.

1. Add TARGETS file
2. Make hipify_python a module as well as a standalone script.

Reviewed By: bddppq

Differential Revision: D10851216

fbshipit-source-id: 5e5c2b1c7fac05c0672c98a8c974636744a6c063
